framework/src/modules/random/utils.ts

```diff
@@ -94,28 +94,24 @@ export const getRandomSeed = (
 		throw new Error('Height is in the future.');
 	}
 
-	for (const seedObject of currentSeeds) {
-		if (seedObject.valid) {
-			randomSeed = bitwiseXOR([randomSeed, seedObject.seedReveal]);
-		}
-	}
-
-	return randomSeed;
+	return bitwiseXOR(currentSeeds);
 };
 
 export const bitwiseXOR = (bufferArray: Buffer[]): Buffer => {
 	if (bufferArray.length === 1) {
 		return bufferArray[0];
 	}
 
-	const bufferSizes = new Set(bufferArray.map(buffer => buffer.length));
-	if (bufferSizes.size > 1) {
-		throw new Error('All input for XOR should be same size');
+	const size = bufferArray[0].length;
+	for (let i = 1; i < bufferArray.length; i += 1) {
```

Comment:
minor point, but we could just loop with `for...of` here

Reply:
Definitely, `for..of` produces a more readable code 👍🏻

Reply:
I'm a big fan of using `for..of` whenever possible, but here I used `for` intentionally. It doesn't need to loop over the first element, as its length is already assigned in `size` constant.